plugwise/helper.py

```diff
@@ -459,9 +477,15 @@ def _appliance_info_finder(self, appliance: etree, appl: Munch) -> Munch:
         if appl.pwclass == "gateway":
             self.gateway_id = appliance.attrib["id"]
             appl.fw = self.smile_version[0]
+            appl.mac = self.smile_mac_address
             appl.model = appl.name = self.smile_name
             appl.v_name = "Plugwise B.V."
 
+            # Adam: check for ZigBee mac address
+            mac_locator = ".//protocols/zig_bee_coordinator/mac_address"
+            if self._domain_objects and self._domain_objects.findall(mac_locator):
+                appl.zigbee_mac = self._domain_objects.findall(mac_locator)[0].text
```

Comment:
In that case, @bouwew, it can be written as (depends on the Python version you guys want to support)

```suggestion
            if self.smile_name == "Adam" and found := self._domain_objects.find(mac_locator
                ".//protocols/zig_bee_coordinator/mac_address"
            ):
                appl.zigbee_mac = found.text
```

Reply:
We'll first have to figure out why the find isn't behaving accordingly and findall does work but I can already rewrite it with findall :)

Reply:
As per Core 3.9 would still be minimal (we ditched testing on 3.8 in the last weeks) - but the construct isn't working with find so lets first fix that